### PR TITLE
issue: 781540 wait for Rx WQE's when closing RING (QP)

### PR DESCRIPTION
**Pull request by rosenbaumalex**
The handling of all FLUSHed WQE's to CQE's takes a different amount of time on each HW.
This commit mark's the last posted WQE. Later, when QP is moved to ERROR state, the
qp_mgr::release_rx_buffers() knows to wait for that marked WR_ID to indicate all CQE's are FLUSHed.

Also, add a test which loops on create and destroy of single bound socket
which creates and releases a VMA RING object. Using 'vma_stats -v 3' you
can follow the RX buffer pool and verify all Rx buffers are returned to
poll as expected.

### CODE REVIEW

**Comment by rosenbaumalex** [comment]
On ConnectX3 flushing all Rx WQE's takes about 1,600 usec.
On ConnectX4 this takes about 23,000 usec. about x20 longer. 
(Tested with CX4Lx FW 14.16.0048)

With new code, even if this flushing process takes longer, VMA will wait for this processing to complete.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/109/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/111/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/112/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/113/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/1/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/117/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/118/ for details (Mellanox internal link).

File: src/vma/dev/qp_mgr.cpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ qp_mgr::qp_mgr(const ring_simple* p_ring, const ib_ctx_handler* p_context, const
 	m_qp(NULL), m_p_ring((ring_simple*)p_ring), m_port_num((uint8_t)port_num), m_p_ib_ctx_handler((ib_ctx_handler*)p_context),
 	m_p_ahc_head(NULL), m_p_ahc_tail(NULL), m_max_inline_data(0), m_max_qp_wr(0), m_p_cq_mgr_rx(NULL), m_p_cq_mgr_tx(NULL),
 	m_rx_num_wr(safe_mce_sys().rx_num_wr), m_tx_num_wr(tx_num_wr), m_rx_num_wr_to_post_recv(safe_mce_sys().rx_num_wr_to_post_recv), 
-	m_curr_rx_wr(0), m_n_unsignaled_count(0), m_n_tx_count(0), m_p_last_tx_mem_buf_desc(NULL), m_p_prev_rx_desc_pushed(NULL),
+	m_curr_rx_wr(0), m_last_posted_rx_wr_id(0), m_n_unsignaled_count(0), m_n_tx_count(0), m_p_last_tx_mem_buf_desc(NULL), m_p_prev_rx_desc_pushed(NULL),
```
Comment (OphirMunk) [inline src/vma/dev/qp_mgr.cpp]: Better initialize pointers with NULL rather than 0

File: src/vma/dev/cq_mgr.cpp [hunk 2]
```diff
@@ -873,6 +878,9 @@ int cq_mgr::drain_and_proccess(bool b_recycle_buffers /*=false*/)
 					}
 				}
 			}
+			if (p_recycle_buffers_last_wr_id) {
+				*p_recycle_buffers_last_wr_id = (uintptr_t)wce[i].wr_id;
```
Comment (OphirMunk) [inline src/vma/dev/cq_mgr.cpp]: @rosenbaumalex Where do you use the updated value in *p_recycle_buffers_last_wr_id?

**Comment by OphirMunk** [comment]
+1

**Comment by rosenbaumalex** [comment]
let's wait with merge for after VMA-8.0 is formally released

**Pull request merged**